compiler/src/scala/quoted/runtime/impl/QuoteMatcher.scala

```diff
@@ -449,6 +474,26 @@ object QuoteMatcher {
      */
     case OpenTree(tree: Tree, patternTpe: Type, args: List[Tree], env: Env)
 
+    /** The Definitions object */
+    def defn(using Context): Definitions = ctx.definitions
+
+    /*
+     * PR-17567 Remaining TODOs
+     * * [ ] Implicit / Contextual parameters
+     * * [x] Nested Method Types
+     * * [ ] Erased Types
+     */
+    def adaptTypes(tpe: Type)(using Context): Type =
+      new Types.TypeMap {
+          def apply(tp: Types.Type): Types.Type = tp match
+            case tp: MethodType => {
+              val paramInfos = tp.paramInfos map adaptTypes
```

Comment:
Here it would be better to call `tp.paramInfos.map.apply(apply)`. Calling it through `adaptTypes` requires the creation of a new `TypeMap` in each recursion.

Reply:
Fixed, thank you